src/app.rs

```diff
@@ -317,23 +324,103 @@ impl App {
             return Err(ServerError::DuplicateCommitment);
         }
 
-        self.database.insert_new_identity(commitment).await?;
+        self.database
+            .insert_new_identity(commitment, DateTime::from(Utc::now()))
+            .await?;
+
+        Ok(())
+    }
+
+    /// Queues a deletion from the merkle tree.
+    ///
+    /// # Errors
+    ///
+    /// Will return `Err` if identity is already queued, not in the tree, or the
+    /// queue malfunctions.
+    #[instrument(level = "debug", skip(self))]
+    pub async fn delete_identity(&self, commitment: Hash) -> Result<(), ServerError> {
+        // TODO: in the comment, it mentions the it will error if not in the tree,
+        // TODO: decide if we are to do this or not and the best way if so
+
+        // TODO: update this to check if there are deletion provers
+        if !self.identity_manager.has_provers().await {
+            warn!(
+                ?commitment,
+                "Identity Manager has no provers. Add provers with /addBatchSize request."
+            );
+            return Err(ServerError::NoProversOnIdInsert);
+        }
+
+        let identity_exists = self.database.identity_exists(commitment).await?;
```

Comment:
Something to keep in mind is that this function simply checks for the existance of the given commitment in the identities table.

But with the new schema an identity could have been already deleted.